theia-extensions/viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx

```diff
@@ -77,9 +62,15 @@ export class TraceExplorerWidget extends BaseWidget {
         layout.addWidget(this.placeholderWidget);
         layout.addWidget(this.traceViewsContainer);
         this.node.tabIndex = 0;
+        signalManager().on(Signals.OPENED_TRACES_UPDATED, () => this.update());
         this.update();
     }
 
+    dispose(): void {
+        super.dispose();
+        signalManager().off(Signals.OPENED_TRACES_UPDATED, () => this.update());
```

Comment:
The listener does not get removed because it is not the same instance being passed that was added earlier.